embedded-service/src/ipc/deferred.rs

```diff
@@ -0,0 +1,98 @@
+//! Definitions for deferred execution of commands
+use core::{
+    marker::PhantomData,
+    sync::atomic::{AtomicU32, Ordering},
+};
+
+use embassy_sync::{blocking_mutex::raw::RawMutex, mutex::Mutex, signal::Signal};
+
+use crate::debug;
+
+/// A unique identifier for a particular command invocation
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct InvocationId(u32);
```

Comment:
Doesn't need to be public
```suggestion
struct InvocationId(u32);
```

Reply:
Done.